lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -837,19 +889,80 @@ std::string SessionState::get_session_id() const {
   return session_id_;
 }
 
-SessionConfig SessionState::get_config() const {
-  return config_;
+std::string SessionState::get_subscriber_id() const {
+  return config_.common_context.sid().id();
 }
 
-uint32_t SessionState::get_local_teid() const {
-  return local_teid_;
+uint32_t SessionState::get_pdu_id() const {
+  return config_.rat_specific_context.m5gsm_session_context().pdu_session_id();
+}
+
+SessionConfig SessionState::get_config() const {
+  return config_;
 }
 
-void SessionState::set_local_teid(
-    uint32_t teid, SessionStateUpdateCriteria& uc) {
-  local_teid_              = teid;
-  uc.is_local_teid_updated = true;
-  uc.local_teid_updated    = teid;
+uint32_t SessionState::get_upf_local_teid() const {
+#if 0
+  char teidar[10];
+  union mix_t {
+    std::uint32_t theDWord;
+    std::uint8_t theBytes[4];
+  };
+  mix_t aVal;
+
+  // Retrieve value
+  std::string teids = config_.rat_specific_context.m5gsm_session_context()
+                          .upf_endpoint()
+                          .teid();
+  uint32_t len = teids.length();
+  memset(teidar, '\0', 10);
+  memcpy(&teidar[0], &teids[0], len);
+  bool byte_zero_skip = true;
+  for (int i = 0; i < 4; ++i) {
+    if (byte_zero_skip && (!teidar[i])) continue;
+    // Rest continue;
+    byte_zero_skip   = false;
+    aVal.theBytes[i] = uint32_t(teidar[i]);
+  }
+  return ntohl(aVal.theDWord);
+#endif
+  return config_.rat_specific_context.m5gsm_session_context()
+      .upf_endpoint()
+      .teid_value();
+}
+
+void SessionState::set_upf_teid_endpoint(
+    std::string ip_addr, uint32_t teid, SessionStateUpdateCriteria& uc) {
```

Comment:
`ip_addr` can be passed as const reference

Reply:
Adopted the change.